### PR TITLE
removed unnecessary gui mount fs checks

### PR DESCRIPTION
**Pull request by badri-pathak**
## Pull request checklist

```release-notes

```

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature Enhancement
- [ ] Test Automation 
- [ ] Code Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Community Operator listing 
- [ ] Other (please describe): 

## What is the current behavior?

-

## What is the new behavior?

- Changes related to removal of mounting checks on the gui nodes before volume_provisioning. Those are not required, if the mount is not present then error comes as part of response. so unnecessary checks not required.

## How risky is this change?
- [ ] Small, isolated change
- [x] Medium, requires regression testing
- [ ] Large, requires functional and regression testing

### CODE REVIEW

**Comment by badri-pathak** [comment]
Image: quay.io/badri_pathak/ibm-spectrum-scale-csi-driver:guimountcheck_test7

**Review by Jainbrt — DISMISSED** [review]

File: driver/csiplugin/controllerserver.go [hunk 1]
```diff
@@ -1576,21 +1565,6 @@ func (cs *ScaleControllerServer) validateSnapId(ctx context.Context, scaleVol *s
...
-				 if err != nil {
-					 return status.Error(codes.Internal, fmt.Sprintf("error in getting filesystem mount details for %s", sourcesnapshot.FsName))
-				 }
-				 if !isFsMounted {
-					 return status.Error(codes.Internal, fmt.Sprintf("filesystem %s is not mounted on GUI node", sourcesnapshot.FsName))
-				 }*/
-		err = checkFilesystemMountOnGUI(ctx, conn, sourcesnapshot.FsName, "", "")
-
-		if err != nil {
-			return err
```
Comment (Jainbrt) [inline driver/csiplugin/controllerserver.go:1590]: if we have removed the call for checkFilesystemMountOnGUI from all the places, should we remove the function definition too ?

**Comment by badri-pathak** [comment]
Image: quay.io/badri_pathak/ibm-spectrum-scale-csi-driver:guimountcheck_v10

File: driver/csiplugin/controllerserver.go (hunk 1 above)
Comment (badri-pathak) [reply driver/csiplugin/controllerserver.go:1590]: Yes, we have removed as it's not being used now.

File: driver/csiplugin/controllerserver.go [hunk 2]
```diff
@@ -961,23 +951,6 @@ func (cs *ScaleControllerServer) checkFileSetLink(ctx context.Context, connector
 	return nil
 }
 
-func checkFilesystemMountOnGUI(ctx context.Context, connector connectors.SpectrumScaleConnector, fsName string, scaleVolName string, primaryOrNone string) error {
-	loggerId := utils.GetLoggerId(ctx)
-	// primary filesytem must be mounted on GUI node so that we can create the softlink
-	// skip if primary and volume filesystem is same
-	klog.V(4).Infof("[%s] volume:[%v] - check if %s filesystem [%v] is mounted on GUI node of %s cluster", loggerId, scaleVolName, primaryOrNone, fsName, primaryOrNone)
-	isFsMounted, err := connector.IsFilesystemMountedOnGUINode(ctx, fsName)
```
Comment (Jainbrt) [inline driver/csiplugin/controllerserver.go:969]: IsFilesystemMountedOnGUINode is also obsolete, right ?

**Review by saurabhwani5 — APPROVED** [review]
PR is working as expected

File: driver/csiplugin/controllerserver.go (hunk 2 above)
Comment (saurabhwani5) [reply driver/csiplugin/controllerserver.go:969]: @badri-pathak , can you please check on this

Comment (badri-pathak) [reply driver/csiplugin/controllerserver.go:969]: @saurabhwani5 @Jainbrt This is a part of gui rest call, written as a Interface, so will be keeping, might be used later. This won't come a lint issue.

**Pull request merged**